app/main.py

```diff
@@ -1,6 +1,20 @@
-from typing import Callable
+from typing import Callable, Any
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = dict()
+
+    @wraps(func)
+    def wrapper(*args) -> Any:
+        name_of_function = func.__name__
+        if name_of_function not in cache_dict:
+            cache_dict[name_of_function] = dict()
+        if args not in cache_dict[name_of_function]:
+            cache_dict[name_of_function][args] = func(*args)
```

Comment:
how about to make a tuple from function name and args, and make it a key in the cache_dict? It will help avoid nested dictionaries